src/transformers/models/whisper/generation_whisper.py

```diff
@@ -745,6 +719,50 @@ def generate(
         if return_segments:
             return {"sequences": sequences, "segments": final_segments}
 
+        if is_shortform:
+            # add eos token:
```

Comment:
Probably possible to keep the EOS token directly in the generated sequence as suggested in the previous comment